CONTRIBUTE.md

```diff
@@ -1,6 +1,34 @@
 # Contribute
 
-How to hack:
+## Issue
+
+Before writing any code, please create an issue first that describes the problem
+you are trying to solve with alternatives that you have considered. A little bit
+of prior communication can save a lot of time on coding. Keep the problem as
+small as possible. If there are two problems, make two issues. We discuss the
+issue and if we reach an agreement on the approach, it's time to move on to a
+PR.
+
+## PR
+
+Follow up with a pull request Post a corresponding PR with the smallest change
+possible to address the issue. Then we discuss the PR, make changes as needed
+and if we reach an agreement, the PR will be merged.
+
+Please do not use `git push --force` on your PR branch for the following reasons:
```

Comment:
I find that sometimes I've done something that only a force push can fix. Not saying we should not ask for this. More noticing this, and maybe it is about some gap in my Git-fu.